### ISSUE TITLE
FALSE POSITIVE: NP_ALWAYS_NULL reports inside a fresh empty-list branch

### ISSUE THREAD

**Issue by chisa22** [spotbugs/spotbugs#4270]
Hi, I found a false positive in SpotBugs 4.10.3 when the fresh ArrayList is empty and does not escape before isEmpty().

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs NP_ALWAYS_NULL`

**Minimal reproducer**

```java
import java.util.ArrayList;

class SpotBugsEmptyList {
  void test() {
    ArrayList<Object> values = new ArrayList<>();
    if (!values.isEmpty()) {
      Object object = null;
      object.toString();
    }
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-np-always-null-fp-empty-list.java
spotbugs -textui -effort:max -low -visitors FindNullDeref classes
```

**Current behavior**

SpotBugs reports `NP_ALWAYS_NULL` at line 8.

**Expected behavior**

SpotBugs should not report `NP_ALWAYS_NULL` because the fresh ArrayList is empty and does not escape before isEmpty().
